torch/csrc/serialization.h

```diff
@@ -13,4 +13,16 @@ ssize_t doRead(io fildes, void* buf, size_t nbytes);
 template <class io>
 ssize_t doWrite(io fildes, void* buf, size_t nbytes);
 
+template <>
+ssize_t doRead<int>(int fildes, void* buf, size_t nbytes);
+
+template <>
+ssize_t doRead<PyObject*>(PyObject* fildes, void* buf, size_t nbytes);
+
+template <>
+ssize_t doWrite<int>(int fildes, void* buf, size_t nbytes);
+
+template <>
+ssize_t doWrite<PyObject*>(PyObject* fildes, void* buf, size_t nbytes);
```

Comment:
What was the error you got when you didn't explicitly list instantiations of this template?

Reply:
It only happens on clang on mac:
![image](https://user-images.githubusercontent.com/5652049/37038555-1e7a61e6-2123-11e8-80ee-93e0e4c94f44.png)
(and analogous errors for doRead)

Reply:
Oh but those errors are only because you use the templates before you declare them. Make sure you instantiate them explicitly before you include the generic header/

Reply:
Oh good point, thank you for catching that @apaszke! I will try that